web-common/src/features/templates/table/TableTemplate.svelte

```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
+  import { createPivotDataStore } from "@rilldata/web-common/features/dashboards/pivot/pivot-data-store";
+  import {
+    PivotChipType,
+    PivotDataStore,
+    PivotDataStoreConfig,
+    PivotState,
+  } from "@rilldata/web-common/features/dashboards/pivot/types";
+  import { createStateManagers } from "@rilldata/web-common/features/dashboards/state-managers/state-managers";
+  import TableRenderer from "@rilldata/web-common/features/templates/table/TableRenderer.svelte";
+  import { TableProperties } from "@rilldata/web-common/features/templates/types";
+  import { V1ComponentSpecRendererProperties } from "@rilldata/web-common/runtime-client";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import { useQueryClient } from "@tanstack/svelte-query";
+  import { Readable, writable } from "svelte/store";
+  import { getTableConfig, hasValidTableSchema } from "./selector";
+
+  export let rendererProperties: V1ComponentSpecRendererProperties;
+
+  const queryClient = useQueryClient();
+  const TABLE_PREFIX = "_custom-table";
+
+  $: instanceId = $runtime.instanceId;
+
+  $: tableProperties = rendererProperties as TableProperties;
+
+  $: tableSchema = hasValidTableSchema(instanceId, tableProperties);
+
+  $: isValidSchema = $tableSchema.isValid;
+
+  $: colDimensions = tableProperties.col_dimensions || [];
+  $: rowDimensions = tableProperties.row_dimensions || [];
+
+  $: pivotState = writable<PivotState>({
+    active: true,
+    columns: {
+      measure: tableProperties.measures.map((measure) => ({
+        id: measure,
+        title: measure,
+        type: PivotChipType.Measure,
+      })),
+      dimension: colDimensions.map((dimension) => ({
+        id: dimension,
+        title: dimension,
+        type: PivotChipType.Dimension,
+      })),
+    },
+    rows: {
+      dimension: rowDimensions.map((dimension) => ({
+        id: dimension,
+        title: dimension,
+        type: PivotChipType.Dimension,
+      })),
+    },
+    expanded: {},
+    sorting: [],
+    columnPage: 1,
+    rowPage: 1,
+    enableComparison: false,
+    rowJoinType: "nest",
+  });
+
+  let pivotDataStore: PivotDataStore | undefined = undefined;
+  let pivotConfig: Readable<PivotDataStoreConfig> | undefined = undefined;
+  $: if (isValidSchema) {
+    const stateManagerContext = createStateManagers({
+      queryClient,
+      metricsViewName: tableProperties.metric_view,
+      extraKeyPrefix: TABLE_PREFIX,
+    });
+
+    pivotConfig = getTableConfig(instanceId, tableProperties, $pivotState);
+    pivotDataStore = createPivotDataStore(stateManagerContext, pivotConfig);
+  }
```

Comment:
Writing down what we discussed [in Slack](https://rilldata.slack.com/archives/C04AGHL77PS/p1722346431806149?thread_ts=1722321321.527189&cid=C04AGHL77PS) – we likely shouldn't re-create the `pivotDataStore` every time the `tableProperties` change, but rather we should update the store's values.

Reply:
Will address in follow up